notes_app/notes/tests.py

```diff
@@ -155,3 +155,291 @@ def test_delete_detail(self):
+            data=self.patch_data,
+            authentication=self.get_credentials()
+        )
+        self.assertHttpAccepted(resp)
+        self.assertEqual(User.objects.get(id=self.api_user_id).last_name, 'testPatch')
+
+    def test_patch_detail_auth_fail(self):
+        """Users that fail Authorization can't PATCH"""
+        resp = self.api_client.put(
+            self.api_detail,
+            format='json',
+            data=self.patch_data,
+            authentication=self.get_wrong_credentials()
+        )
+        original_lname = User.objects.get(id=self.api_user_id).last_name
+        self.assertHttpUnauthorized(resp)
+        self.assertNotEqual(User.objects.get(id=self.api_user_id).last_name, 'testPatch')
+        self.assertEqual(User.objects.get(id=self.api_user_id).last_name,  original_lname)
+
+    def test_delete_detail(self):
+        total_count = User.objects.all().count()
+        self.assertEqual(User.objects.count(), total_count)
+        self.assertHttpAccepted(
+            self.api_client.delete(
+                self.api_detail,
+                format='json',
+                authentication=self.get_credentials()
+            )
+        )
+        self.assertEqual(User.objects.count(), (total_count - 1))
+
+    # 401 Unauthorized
+    def test_unauthorized(self):
+        """Unauthorized Users cant GET, POST list & GET, PUT, PATCH, DELETE detail"""
+        self.assertHttpUnauthorized(
+            self.api_client.get(
+                self.api_list,
+                format='json'
+            )
+        )
+        self.assertHttpUnauthorized(
+            self.api_client.post(
+                self.api_list,
+                format='json',
+                data=self.post_data
+            )
+        )
+        self.assertHttpUnauthorized(
+            self.api_client.get(
+                self.api_detail,
+                format='json'
+            )
+        )
+        self.assertHttpUnauthorized(
+            self.api_client.put(
+                self.api_detail,
+                format='json',
+                data=self.put_data
+            )
+        )
+        self.assertHttpUnauthorized(
+            self.api_client.patch(
+                self.api_detail,
+                format='json',
+                data=self.put_data
+            )
+        )
+        self.assertHttpUnauthorized(
+            self.api_client.delete(
+                self.api_detail,
+                format='json',
+                data=self.put_data
+            )
+        )
+
+    # 405 Method Not Allowed
+    def test_not_allowed(self):
+        """Unauthorized & Authorized Users can't PUT PATCH DELETE list & POST detail"""
+        # PUT list unauth
+        self.assertHttpMethodNotAllowed(
+            self.api_client.put(
+                self.api_list,
+                format='json',
+                data=self.put_data
+            )
+        )
+        # PUT list auth
+        self.assertHttpMethodNotAllowed(
```

Comment:
what's the point of duplicating such test with logged in user and not? It's enough only one, because the same HttpMethodNotAllowed returns